src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php

```diff
@@ -0,0 +1,395 @@
+
+    /** @var string|null */
+    private $defaultLocale;
+
+    public function __construct(string $projectId, string $apiToken, string $organizationDomain = null, HttpClientInterface $client = null, LoaderInterface $loader = null, LoggerInterface $logger = null, string $defaultLocale = null, XliffFileDumper $xliffFileDumper = null)
+    {
+        $this->projectId = $projectId;
+        $this->apiToken = $apiToken;
+        $this->organizationDomain = $organizationDomain;
+        $this->loader = $loader;
+        $this->logger = $logger;
+        $this->defaultLocale = $defaultLocale;
+        $this->xliffFileDumper = $xliffFileDumper;
+
+        parent::__construct($client);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf(CrowdinProviderFactory::SCHEME . '://%s', $this->getEndpoint());
+    }
+
+    public function getName(): string
+    {
+        return CrowdinProviderFactory::SCHEME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function write(TranslatorBag $translatorBag): void
+    {
+        foreach($translatorBag->getDomains() as $domain) {
+            /** @var MessageCatalogue $catalogue */
+            foreach ($translatorBag->getCatalogues() as $catalogue) {
+                $content = $this->xliffFileDumper->formatCatalogue($catalogue, $domain);
+
+                $fileId = $this->getFileId($domain);
+
+                if ($catalogue->getLocale() === $this->defaultLocale) {
+                    if (!$fileId) {
+                        $this->addFile($domain, $content);
+                    } else {
+                        $this->updateFile($fileId, $domain, $content);
+                    }
+                } else {
+                    $this->uploadTranslations($fileId, $domain, $content, $catalogue->getLocale());
+                }
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read(array $domains, array $locales): TranslatorBag
+    {
+        $translatorBag = new TranslatorBag();
+
+        foreach ($domains as $domain) {
+            $fileId = $this->getFileId($domain);
+
+            if (!$fileId) {
+                continue;
+            }
+
+            foreach ($locales as $locale) {
+                if ($locale !== $this->defaultLocale) {
+                    $content = $this->exportProjectTranslations($locale, $fileId);
+                } else {
+                    $content = $this->downloadSourceFile($fileId);
+                }
+
+                $translatorBag->addCatalogue($this->loader->load($content, $locale, $domain));
+            }
+        }
+
+        return $translatorBag;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(TranslatorBag $translatorBag): void
+    {
+        foreach($translatorBag->getDomains() as $domain) {
+            $fileId = $this->getFileId($domain);
+
+            if ($fileId) {
+                $this->deleteFile($fileId);
+            }
+        }
```

Comment:
Oh! This workaround will remove all translations (key + messages) of a domain. This it not that we want. Here, we want to delete only the translations contained in the TranslatorBag passed as argument. Which contains the diff, computed in the Command file.

We have to rework this method